src/IW/Server/Issue.hs

```diff
@@ -0,0 +1,83 @@
+module IW.Server.Issue
+       ( -- * API
+         IssueAPI
+       , issueServer
+
+         -- * Handlers
+       , issueByIdHandler
+       , issuesHandler
+       ) where
+
+import IW.App.Error (WithError)
+import IW.Core.Issue (Issue (..))
+import IW.Db (WithDb, getIssues, getIssueById, getIssuesByLabel)
+import IW.Server.Types (AppServer, ToApi)
+
+data Sort = Id | Title
+  deriving (Generic, Show, Eq)
+
+instance FromHttpApiData Sort where
+    parseQueryParam :: Text -> Either Text Sort
+    parseQueryParam "id"    = Right Id
+    parseQueryParam "title" = Right Title
+    parseQueryParam _       = Left "Error parsing sort value"
+
+data IssueSite route = IssueSite
+    { -- | Fetch issue by issue ID
+      issueByIdRoute :: route
+        :- "issues"
+        :> Capture "id" Int
+        :> Get '[JSON] Issue
+    , issuesRoute :: route
+        :- "issues"
+        :> QueryParam "sort" Sort
+        :> QueryFlag "desc"
+        :> QueryParam "label" Text
+        :> Get '[JSON] [Issue]
+    } deriving (Generic)
+
+type IssueAPI = ToApi IssueSite
+
+issueServer :: IssueSite AppServer
+issueServer = IssueSite
+    { issueByIdRoute  = issueByIdHandler
+    , issuesRoute     = issuesHandler
+    }
+
+issueByIdHandler
+    :: ( WithDb env m
+       , WithError m
+       )
+    => Int
+    -> m Issue
+issueByIdHandler = getIssueById
+
+issuesHandler 
+    :: ( WithDb env m
+       , WithError m
+       )
+    => Maybe Sort
+    -> Bool
+    -> Maybe Text
+    -> m [Issue]
+issuesHandler sortParam isDesc labelParam = do 
+  issues       <- case labelParam of
+                    Nothing -> getIssues
+                    Just p  -> getIssuesByLabel p
+  return $ sortDesc isDesc . sortIssues sortParam $ issues
+  where
+    sortById :: [Issue] -> [Issue]
+    sortById = sortOn issueId
```

Comment:
We decided to do sorting on frontend. But just for future reference:

1. It's better to do sorting in the SQL query directly rather than in memory. Databases can sort faster and more efficiently, and linked lists are not the best data structure for sorting.
2. If you want to sort by some field, it's better to use `sortWith` instead of `sortOn`. `sortOn` creates tuples and sort tuples, so `sortOn` is more efficient if you want to sort by some expensive function, like, length for lists. But if you're just sorting by field then it's better to use `sortWith`.

Reply:
Got it! Thanks for the info.